### DISCUSSION TITLE
4K Video Downloader+

### DISCUSSION THREAD

**Discussion by viking1304** [Homebrew/discussions#4667] [Casks] [answered]
### Output of `brew config`

```shell
Not applicable
```

### Output of `brew doctor`

```shell
Not applicable
```

### Description of issue

4K Video Downloader is now 4K Video Downloader+. As you can see in FAQ, 4K Video Downloader still exists, but new features will only be introduced in 4K Video Downloader+.

4K Video Downloader+ has version 1.0.1.0019, while legacy 4K Video Downloader has version 4.25.1.5490.

What should be done in this situation? Create a new cask for 4K Video Downloader+?

### Relevant casks

4k-video-downloader

**Comment by miccal** [comment]
https://github.com/Homebrew/homebrew-cask/pull/151622

**Comment by krehel** [comment]
> What happens to the old 4K Video Downloader?

>4K Video Downloader still exists, you can use it and all its features just like before. However, new features will only be introduced in 4K Video Downloader+ due to technical reasons.

Given this, even though it's a software that has been updated recently, does it make sense to mark `4k-video-downloader` as technically  discontinued if no more development is going to occur?  And/or a `caveat` with a reference to the new `4k-video-downloaderplus`?

**Comment by viking1304** [comment]
> Given this, even though it's a software that has been updated recently, does it make sense to mark `4k-video-downloader` as technically discontinued if no more development is going to occur? And/or a `caveat` with a reference to the new `4k-video-downloaderplus`?

The situation is indeed a little confusing. They say the old license works with a new app and encourage people to switch to the new app while the old one is still available and working.

The strange thing is that the old app is updated one day after (July 20th) the last release of the new app (July 19th).

It looks like the old app will remain in maintenance-only mode. So we might get new updates, but no new features.


> Do I have to upgrade to 4K Video Downloader+?
You can continue using 4K Video Downloader. But if you want to have access to more features now and others that we will implement in the future, we recommend that you [upgrade to 4K Video Downloader+](https://www.4kdownload.com/license/upgrade/1?source=videodownloader).

> Can I use my old license after the upgrade?
Once you’ve used the license to upgrade to 4K Video Downloader+, it won’t work in 4K Video Downloader. If you want to use both applications, you will need a separate license for each.

Based on those answers, I believe two active casks are the only valid solution.

**Comment by viking1304** [comment, marked as answer]
The new version 4.26.1.5520 of the "legacy" version was released yesterday, so both casks should remain live.
